src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -25,30 +25,25 @@ public class ExtensionSettings {
      */
     public static final String EXTENSION_DESCRIPTOR = "src/test/resources/extension.yml";
 
-    public String getExtensionname() {
-        return extensionname;
+    /**
+     * Jackson requires a default constructor.
+     */
+    public ExtensionSettings() {
+        super();
     }
 
-    public void setExtensionname(String extensionname) {
-        this.extensionname = extensionname;
+    public String getExtensionname() {
```

Comment:
(Style preference/suggestion) I know you're using the existing field names but I think using camelCase may improve readability for this one in particular because it's so long and has the double n's: `extensionName`.  Also consider for `hostAddress` and `hostPort`.